coursera/network.py

```diff
@@ -46,9 +46,12 @@ def get_reply(session, url, post=False, data=None, headers=None):
     try:
         reply.raise_for_status()
     except requests.exceptions.HTTPError as e:
-        logging.error("Error %s getting page %s", e, url)
-        logging.error("The server replied: %s", reply.text)
-        raise
+        if reply.json()['message'] == 'Item is locked':
```

Comment:
`get_reply` is rather a low-level function. It's a bad idea to put that much knowledge about the response message into this function. On top of that, relying on `message` value is a treading on thin ice. You should check error codes, not human-readable messages.